cypress/e2e/tests/pages/fleet/gitrepo.spec.ts

```diff
@@ -1,29 +1,52 @@
 import { GitRepoCreatePo } from '@/cypress/e2e/po/pages/fleet/gitrepo-create.po';
 import { FleetDashboardPagePo } from '@/cypress/e2e/po/pages/fleet/fleet-dashboard.po';
+import { gitRepoCreateRequest } from '@/cypress/e2e/blueprints/fleet/gitrepos';
 
 describe('Git Repo', { tags: '@adminUser' }, () => {
-  beforeEach(() => {
-    cy.login();
-  });
-
   describe('Create', () => {
     let gitRepoCreatePage: GitRepoCreatePo;
     const repoList = [];
 
-    beforeEach(() => {
+    before(() => {
+      cy.login();
       gitRepoCreatePage = new GitRepoCreatePo('local');
-    });
-
-    it('Should be able to create a git repo', () => {
+      cy.interceptAllRequests('POST');
       gitRepoCreatePage.goTo();
-      gitRepoCreatePage.setRepoName('fleet-e2e-test-gitrepo');
-      gitRepoCreatePage.setGitRepoUrl('https://github.com/rancher/fleet-test-data.git');
-      gitRepoCreatePage.setBranchName();
-      gitRepoCreatePage.gitRepoPaths().setValueAtIndex('simple', 0);
+      const { name } = gitRepoCreateRequest.metadata;
+      const {
+        repo, branch, paths, helmRepoURLRegex
+      } = gitRepoCreateRequest.spec;
+
+      gitRepoCreatePage.setRepoName(name);
+      gitRepoCreatePage.setGitRepoUrl(repo);
+      gitRepoCreatePage.selectWorkspace('fleet-default');
+      gitRepoCreatePage.setBranchName(branch);
+      gitRepoCreatePage.helmAuth().createBasicAuth('test', 'test');
+      gitRepoCreatePage.setHelmRepoURLRegex(helmRepoURLRegex);
+
+      gitRepoCreatePage.gitRepoPaths().setValueAtIndex(paths[0], 0);
       gitRepoCreatePage.goToNext();
       gitRepoCreatePage.create();
 
-      repoList.push('fleet-e2e-test-gitrepo');
+      repoList.push(name);
+    });
+
+    it('Should be able to create a git repo', () => {
+      // First request is for creating credentials
+      let secretName = '';
+
+      cy.wait('@interceptAllRequests0').then(({ request, response }) => {
```

Comment:
Would it not be more convenient to return the `cy.intercept()` method, so you could wait without the alias and then take the first request response as array within the arguments?

E.g.:

```ts
cy.interceptAllRequests().then(([{ request, response }]) => {
```

Reply:
As discussed, `cy.intercept()` has to be initialized before the request.